### PR TITLE
Allow gzip/deflate encoding

### PR DESCRIPTION
**Pull request by rfink**
Addresses #59

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -5,10 +5,35 @@
...
+ */
+function parseResponse(res, body, resolve, reject) {
+  try {
+    if (!/application\/json/.test(res.headers['content-type']) || body.trim() === '') {
+      return resolve(body);
+    }
+
+    const json = JSON.parse(body);
+
+    if (Object.prototype.hasOwnProperty.call(json, 'error') || Object.prototype.hasOwnProperty.call(json, 'errors')) {
```
Comment (jmnsf) [inline lib/request.js]: Since this is a plain parsed object, why not check for properties directly? i.e. `if (json.error || json.errors) {`

Comment (rfink) [reply lib/request.js]: Hey, great question. I believe that's how it is in the code currently, that particular piece was just moved. I can change the style if you'd like, no problem. Let me know.

Comment (rfink) [reply lib/request.js]: No problem! Thanks! Done.

**Comment by jmnsf** [comment]
Cheers 🍻

**Comment by rfink** [comment]
Cheers 🍻! Great module, and thanks.

**Pull request merged**